src/containerapp/azext_containerapp/containerapp_auth_decorator.py

```diff
@@ -3,11 +3,110 @@
+            if "platform" not in self.existing_auth:
+                self.existing_auth["platform"] = {}
+            self.existing_auth["platform"]["enabled"] = self.get_argument_enabled()
+
+        if self.get_argument_runtime_version() is not None:
+            if "platform" not in self.existing_auth:
+                self.existing_auth["platform"] = {}
+            self.existing_auth["platform"]["runtimeVersion"] = self.get_argument_runtime_version()
+
+        if self.get_argument_config_file_path() is not None:
+            if "platform" not in self.existing_auth:
+                self.existing_auth["platform"] = {}
+            self.existing_auth["platform"]["configFilePath"] = self.get_argument_config_file_path()
+
+        if self.get_argument_unauthenticated_client_action() is not None:
+            if "globalValidation" not in self.existing_auth:
+                self.existing_auth["globalValidation"] = {}
+            self.existing_auth["globalValidation"]["unauthenticatedClientAction"] = self.get_argument_unauthenticated_client_action()
+
+        if self.get_argument_redirect_provider() is not None:
+            if "globalValidation" not in self.existing_auth:
+                self.existing_auth["globalValidation"] = {}
+            self.existing_auth["globalValidation"]["redirectToProvider"] = self.get_argument_redirect_provider()
+
+        if self.get_argument_excluded_paths() is not None:
+            if "globalValidation" not in self.existing_auth:
+                self.existing_auth["globalValidation"] = {}
+            self.existing_auth["globalValidation"]["excludedPaths"] = self.get_argument_excluded_paths().split(",")
+
+        self.existing_auth = update_http_settings_in_auth_settings(self.existing_auth, self.get_argument_require_https(),
+                                                                   self.get_argument_proxy_convention(), self.get_argument_proxy_custom_host_header(),
+                                                                   self.get_argument_proxy_custom_proto_header())
+        
     def construct_payload(self):
-        super().construct_payload()
+        self.parent_construct_payload()
+        self.set_up_token_store()
+
+    def set_up_token_store(self):
+        if self.get_argument_token_store() is None:
+            return
+
+        if self.get_argument_token_store() is False:
+            safe_set(self.existing_auth, "login", "tokenStore", "enabled", value=False)
+            return
+
+        safe_set(self.existing_auth, "login", "tokenStore", "enabled", value=True)
+        safe_set(self.existing_auth, "login", "tokenStore", "azureBlobStorage", value={})
+
+        param_provided = 0
+        if self.get_argument_sas_url_secret() is not None:
+            param_provided += 1
+        if self.get_argument_sas_url_secret_name() is not None:
+            param_provided += 1
+        if self.get_argument_blob_container_uri() is not None:
+            param_provided += 1
```

Comment:
```suggestion
        param_provided = sum(1 for param in [self.get_argument_sas_url_secret(), self.get_argument_sas_url_secret_name(), self.get_argument_blob_container_uri()] if param is not None)

```

Reply:
Hi @njuCZ 
I think this validation might have problem, what if the customer input `--blob-container-uri` and `--blob-container-identity`? Or if customer only input `--blob-container-identity`?

Reply:
--blob-container-identity default is empty, which use system assigned managed identity.
for now, when user input --blob-container-identity only, the validation will report error